lib/routelib/routelib.lua

```diff
@@ -1155,15 +1155,17 @@ local function route_zfailover_f(rctx, arg)
         rctx:enqueue(r, far)
         rctx:wait_cond(farcount, mode)
 
-        -- look for a good result, else any OK, else any result.
+        -- wait for a good result if miss==true
+        -- if miss==false, return OK or GOOD, whatever comes first
+        -- return last error if no good or ok comes in
         local final = nil
         for x=1, #far do
             local res, tag = rctx:result(far[x])
             if tag == mcp.RES_GOOD then
                 return res
-            elseif tag == mcp.RES_OK then
-                final = res
-            elseif final ~= nil then
+            elseif tag == mcp.RES_OK and miss == false then
```

Comment:
Think this will return a worse result sometimes:
- We're waiting with `WAIT_FASTGOOD` - which resumes immediately if a HIT is found.
- We need to look through all results for the "best" result, since any of them could be GOOD
- Thus if the first result is OK but the second one is GOOD this code would return a miss?